### ISSUE TITLE
Recomment FreeOTP instead of proprietary google authenticator

### ISSUE THREAD

**Issue by ian-kelling** [tootsuite/mastodon#14084]

**Comment by ian-kelling** [comment]
This is in config/locales/en.yml and related files. This is like the old banners that said "works best in Internet Explorer." Comon now, Mastodon is free software, lets promote free software. Also, I recommend saying "second device" instead of phone, because you can use oathtool on GNU/Linux, and be even more free than almost ever phone.

This is a comment because I accidentally hit enter and can't edit the issue without running nonfree js or learning some crazy api call.

**Comment by ariasuni** [comment]
Relevant string (`two_factor_authentication.instructions_html`):

> `<strong>Scan this QR code into Google Authenticator or a similiar TOTP app on your phone</strong>. From now on, that app will generate tokens that you will have to enter when logging in.`

(there’s a typo on “similar” that could be fixed at the same time)

FreeOTP isn’t maintained anymore, but the two forks andOTP and FreeOTP+ are. It could be recommended before or instead Google Authenticator, though it should be noted that Google Authenticator is recommended because it’s probably what a lot of users already know, and it is available for iOS and others. 

After a bit of thinking, I guess this can be a good replacement for the current string:

> Scan this QR code into a one-time password application on another device. You can use e.g. FreeOTP+ (Android), or Google Authenticator (multi-platform). From now on, that app will generate tokens that you will have to enter when logging in.

**Comment by Sasha-Sorokin** [comment]
> Comon now, Mastodon is free software, lets promote free software.

I don't see this as a good argument, neither end user cares if they aren't geeky enough. Being free or open source software doesn't magically make it the best software in the world, gonna face the truth. It's the experience that does it.

If we're targeting common audience, they most probably would use phone (running iOS or Android) for their authenticator app, which is either Microsoft Authenticator or Google Authenticator (both are multi-platform).

If we still going to choose which app to “promote”, andOTP seems to be a little friendlier to the end user, while providing much better options. We shouldn't really overcomplicate this string with the names of all authenticator apps in the world and platforms they work on.

> \<strong\>Scan this QR code into Microsoft Authenticator, andOTP or a similar TOTP app on your phone\</strong\>. From now on, that app will generate tokens that you will have to enter when logging in.

Even better, let's consider making “a similar TOTP app” a link to documentation instead on what is 2FA is and how could you set up (e.g. “Secure your account”), but first we need such article.

**Comment by SuperSandro2000** [comment]
> Being free or open source software doesn't magically make it the best software in the world

Depends on what you value but free as in freedom and free as in free beer is always preferable even if it lacks some features.

Google Authenticator was dead for many years, has no password protection, export functionality or customization.

Also my pitch on the text:

> Scan this QR code into a two factor authentication (2FA) application on another device. Some apps that support this are FreeOTP+ (Android), andOTP (Android) or Google Authenticator (Android, iOS). From now on, that app will generate tokens that you will have to enter when logging in.

Google Authenticator is not multiplatform. It does not support Linux, Mac or Windows.

**Comment by ariasuni** [comment]
> Google Authenticator is not multiplatform. It does not support Linux, Mac or Windows.

Multi-platform just means several platforms. Not sure if it’s up-to-date, but Wikipedia say it’s also available for Blackberry OS, and it seemed better to me to say “multi-platform” than to list “Android, iOS, Blackberry OS” in an already long enough paragraph.

**Comment by SuperSandro2000** [comment]
> Blackberry OS

is dead.

It still does not support Windows, Mac or Linux which Multi-platform implies.

**Comment by ryliejamesthomas** [comment]
Small note: when using initialisms you should be using the `<abbr>` element, ideally with the `title` attribute.

I think @ariasuni 's rewrite does a good job of not overloading the reader with tech stuff.

**Comment by manuelviens** [comment]
Maybe a list of clients and their plateforms?

I use Aegis from F-Droid.

**Comment by ariasuni** [comment]
I don’t think we want something complex and that needs to be kept up-to-date frequently. Thinks translations, bikeshedding, scary looking block of text.

If there is a good external resources for listing OTP app, it could be used, but I couldn’t find something like that.
